catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -552,139 +548,100 @@ protected ResourceInfo getResourceInfo(ResourceRequest resourceRequest, String s
-                            null,
-                            true,
-                            fanoutEnabled);
-                    if (!queryResponse.getResults()
-                            .isEmpty()) {
-                        metacard = queryResponse.getResults()
-                                .get(0)
-                                .getMetacard();
-                        resourceUri = metacard.getResourceURI();
-                        federatedSite.append(metacard.getSourceId());
-                        LOGGER.debug("Trying to lookup resource URI {} for metacardId: {}",
-                                resourceUri,
-                                metacardId);
-                    } else {
-                        throw new ResourceNotFoundException(
-                                "Could not resolve source id for URI by doing an id based query: "
-                                        + metacardId);
-                    }
-
-                    if (!requestProperties.containsKey(Metacard.ID)) {
-                        requestProperties.put(Metacard.ID, metacardId);
-                    }
-                    if (!requestProperties.containsKey(Metacard.RESOURCE_URI)) {
-                        requestProperties.put(Metacard.RESOURCE_URI, resourceUri);
-                    }
-                } else {
-                    throw new ResourceNotSupportedException(
-                            "The GetResourceRequest with attribute value of class '"
-                                    + value.getClass()
-                                    + "' is not supported by this instance of the CatalogFramework.");
-                }
-            } else {
-                throw new ResourceNotSupportedException(
-                        "The GetResourceRequest with attribute name '" + name
-                                + "' is not supported by this instance of the CatalogFramework.");
+                String metacardId = (String) value;
+                LOGGER.debug("metacardId = {},   site = {}", metacardId, site);
+                query = createMetacardIdQuery(metacardId);
             }
+
+            QueryRequest queryRequest = new QueryRequestImpl(anyTag(query,
+                    site,
+                    isEnterprise),
+                    isEnterprise,
+                    Collections.singletonList(site == null ? this.getId() : site),
+                    resourceRequest.getProperties());
+
+            resourceInfo = getResourceInfo(queryRequest,
+                    requestProperties,
+                    federatedSite,
+                    fanoutEnabled);
         } catch (UnsupportedQueryException | FederationException e) {
 
             throw new ResourceNotFoundException(DEFAULT_RESOURCE_NOT_FOUND_MESSAGE, e);
         }
 
-        LOGGER.debug("Returning resourceURI: {}", resourceUri);
-        if (resourceUri == null) {
+        LOGGER.debug("Returning resourceURI: {}", resourceInfo.getResourceUri());
```

Comment:
❓ Does it make any sense to log `Returning resourceURI: null` right before throwing the exception like this?